events/src/main/java/com/app/events/serviceimpl/TicketServiceImpl.java

```diff
@@ -82,12 +87,30 @@ public Ticket reserveTicket(Long id, Long userId, Long ticketVersion) throws Exc
 	}
 
 	@Override
-	public Ticket buyTicket(Long id, Long userId) throws Exception {
+	public Map<String,Object> ticketPaymentCreation(Long id, Long userId) throws Exception{
+
 		Ticket ticketToUpdate = findOne(id);
+		if(ticketToUpdate.getTicketState().equals(TicketState.BOUGHT) && (ticketToUpdate.getUser().getId() == userId))
```

Comment:
fali provera da li je rezervisan od strane nekog drugog korisnika, a ako je vec kupljena moze da baci neki exception, nek ne vraca null, to nista ne znaci